### PR TITLE
[release-1.25] When crio restarts, restore the infraContainer

### PR DESCRIPTION
**Pull request by kwilczynski**
This is a manual cherry-pick of https://github.com/cri-o/cri-o/pull/7726

/kind bug
/assign kwilczynski

```release-note
Restore infra containers state on CRI-O restart. Without this, the infra containers will be accounted as missing, leading to a spurious error message.
```

### CODE REVIEW

File: internal/lib/container_server_test.go
Comment (kwilczynski) [inline internal/lib/container_server_test.go:169]: **To the reviewer:**

Updated for CRI-O 1.25 to resolve the following function signature difference:

```console
# github.com/cri-o/cri-o/internal/lib_test [github.com/cri-o/cri-o/internal/lib.test]
./container_server_test.go:169:55: too many arguments in call to sut.GetInfraContainer
	have ("context".Context, string)
	want (string)
```

File: internal/lib/container_server.go
Comment (kwilczynski) [inline internal/lib/container_server.go:285]: **To the reviewer:**

Updated for CRI-O 1.25 to resolve the following function signature difference:

```console
# github.com/cri-o/cri-o/internal/lib
Error: internal/lib/container_server.go:285:27: too many arguments in call to c.AddInfraContainer
	have (context.Context, *oci.Container)
	want (*oci.Container)
```

**Comment by kwilczynski** [comment]
@cri-o/ocicni-maintainers, please have a look. Thank you!

**Comment by kwilczynski** [comment]
@cri-o/cri-o-maintainers, please have a look. Thank you!

**Review by saschagrunert — APPROVED** [review]
/hold

Feel free to unhold when the backport is ready to go.

**Pull request merged**
